src/main/java/org/springframework/retry/backoff/BackOffPolicyBuilder.java

```diff
@@ -217,7 +218,7 @@ public BackOffPolicyBuilder randomSupplier(Supplier<Boolean> randomSupplier) {
 	public BackOffPolicy build() {
 		if (this.multiplier != null && this.multiplier > 0 || this.multiplierSupplier != null) {
 			ExponentialBackOffPolicy policy;
-			if (Boolean.TRUE.equals(this.random)) {
+			if (isRandom()) {
```

Comment:
I see that `randomSupplier` is ignored in this class and apparently it cannot be a runtime option anyway: we have to determine a `BackOffPolicy` instance during this `build()` call.
So, I find the fix for this class as good one, but probably `@Backoff(randomExpression)` Javadocs need to be fixed that this expression is always evaluated at configuration phase.

Reply:
@artembilan as suggested, javadoc has been fixed. Please have a look.